src/main/java/botmanager/BotManager.java

```diff
@@ -0,0 +1,73 @@
+package botmanager;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+import action.Action;
+import enums.Command;
+import exception.BotException;
+import task.TaskList;
+
+/**
+ * Entry point for the BotManager Chatbot.
+ *
+ * @author dwsc37
+ */
+public class BotManager {
+    private final Ui ui;
+    private final Parser parser;
+    private final Storage storage;
+
+    /**
+     * Constructor for a BotManager.
+     * Initialises the helper classes needed.
+     */
+    public BotManager() {
+        ui = new Ui();
+        parser = new Parser();
+        storage = new Storage();
+    }
+
+    /**
+     * Runs BotManager.
+     */
+    public void run() {
+        TaskList taskList = new TaskList();
+        try {
+            storage.loadTaskList(taskList);
+        } catch (FileNotFoundException e) {
+            ui.showLoadError();
+            storage.initFile();
+        }
+        ui.start();
+        while (true) {
+            // read user input
+            String input = ui.readUserInput();
+            if (input.equals(Command.EXIT.getInput())) {
```

Comment:
The nested function calls might be a bit confusing to read and understand. Perhaps extracting out `Command.EXIT.getInput()` might improve readability?